core/AssetManager/UIAsset/OnDiskUIAsset.php

```diff
@@ -58,12 +59,18 @@ public function delete()
     {
         if ($this->exists()) {
 
-            if (!unlink($this->getAbsoluteLocation()))
+            try {
+                Filesystem::remove($this->getAbsoluteLocation());
+            } catch (Exception $e) {
                 throw new Exception("Unable to delete merged file : " . $this->getAbsoluteLocation() . ". Please delete the file and refresh");
+            }
 
             // try to remove compressed version of the merged file.
-            @unlink($this->getAbsoluteLocation() . ".deflate");
-            @unlink($this->getAbsoluteLocation() . ".gz");
+            try {
+                Filesystem::remove($this->getAbsoluteLocation() . ".deflate");
+                Filesystem::remove($this->getAbsoluteLocation() . ".gz");
+            } catch (Exception $e) {
+            }
```

Comment:
Surrounding the calls in a try-catch every time we want to ignore unlink failures seems excessive... Maybe instead there can be two methods 'tryRemove' and 'remove'? tryRemove would do the same logic as remove now but return true/false based on whether remove succeeds or not. remove would use tryRemove and if false, throws.

What do you think?